tests/devices/qubit/test_preprocess.py

```diff
@@ -260,6 +252,33 @@ def test_expand_fn_non_commuting_measurements(self):
         new_qs = expand_fn(qs)
         assert new_qs.measurements == qs.measurements
 
+    @pytest.mark.parametrize(
+        "prep_op", (qml.BasisState([1], wires=0), qml.QubitStateVector([0, 1], wires=1))
+    )
+    def test_expand_fn_state_prep(self, prep_op):
+        """Test that the expand_fn only expands mid-circuit instances of StatePrep"""
+        ops = [
+            prep_op,
+            qml.Hadamard(wires=0),
+            qml.QubitStateVector([0, 1], wires=1),
+            qml.BasisState([1], wires=0),
+            qml.RZ(0.123, wires=1),
+        ]
+        measurements = [qml.expval(qml.PauliZ(0)), qml.probs()]
+        tape = QuantumScript(ops=ops, measurements=measurements)
+
+        expanded_tape = expand_fn(tape)
+        expected = [
+            prep_op,
```

Comment:
Why it is not decomposed if it is the first operation?

Reply:
If it's the first operation, we just update the state vector of the simulator itself